packages/next-server/server/render.tsx

```diff
@@ -344,7 +344,19 @@ export async function renderToHTML(
   )
 
   try {
-    props = await loadGetInitialProps(App, { Component, router, ctx })
+    props = await loadGetInitialProps(App, {
+      Component,
+      AppTree: (props: any) => {
+        const appProps = { Component, router, ...props }
```

Comment:
Reason you can remove them is that you're providing them here. Also it should be move the other way around:

```suggestion
        const appProps = { ...props, Component, router }
```

Otherwise users can override Component and router which is something we do not want.

Reply:
Yup, missed a commit